e2e/cypress/integration/ui/settings/project_mgmt.spec.ts

```diff
@@ -75,171 +77,190 @@ describe('project management', () => {
 
     cy.get('[data-cy=update-project-name]').focus().clear()
-      .type(updatedProjectName, { delay: typeDelay })
+      .type(updatedProjectName)
       .should('have.value', updatedProjectName);
     cy.get('app-project-details chef-button').contains('Save').click();
 
     cy.get('app-project-details h1.page-title').contains(updatedProjectName);
   });
 
-  itFlaky('can delete a project rule', () => {
-    cy.get('app-project-details').contains('Edits are pending');
-    cy.get('[data-cy=rules-tab]').click();
+  it('cannot delete a project with rules', () => {
+    const rule: Rule = {
+      id: `${cypressPrefix}-rule`,
+      name: 'test rule',
+      project_id: customWithoutPolsProjectID,
+      type: 'NODE',
+      conditions: [
+        {
+          attribute: 'CHEF_ORGANIZATION',
+          operator: 'EQUALS',
+          values: ['foo']
+        }
+      ]
+    };
+    // add rule via the API to save time,
+    // since we test rules more thoroughly in project_rule_mgmt.spec.ts
+    cy.request({
+      auth: { bearer: adminIdToken },
+      method: 'POST',
+      url: `/apis/iam/v2/projects/${customWithoutPolsProjectID}/rules`,
+      body: rule
+    }).then((response) => {
+      expect(response.status).to.equal(200);
+    });
+
+    // navigate back to the project list page and get latest project data
+    cy.route('GET', '/apis/iam/v2/projects').as('getProjects');
+    cy.get('.breadcrumb').click();
+    cy.wait('@getProjects');
+
+    cy.get('app-project-list chef-td').contains(customWithoutPolsProjectID).parent()
+      .find('.mat-select-trigger').as('controlMenu');
 
-    cy.get('app-project-details chef-td').contains(ruleID).parent()
-      .find('mat-select').as('controlMenu');
-    // we throw in a should so cypress retries until introspection allows menu to be shown
-    cy.get('@controlMenu').should('be.visible')
+    // we throw in a `should` so cypress retries until introspection allows menu to be shown
+    cy.get('@controlMenu').scrollIntoView().should('be.visible')
+      .click();
+    cy.get('[data-cy=delete-project]').should('be.visible')
       .click();
-    cy.get('@controlMenu').find('[data-cy=delete-rule]').click({ force: true });
 
-    cy.get('app-project-details chef-button').contains('Delete Rule').click();
+    cy.get('app-project-list app-message-modal')
+      .contains('Could Not Delete Project').should('be.visible');
 
-    // since this is a cypress custom project, we know this is the only rule.
-    // the empty UI should show up so entire table will be missing.
-    cy.get('app-project-details chef-tbody').should('not.exist');
-    cy.get('app-project-details').contains('Edits are pending').should('not.exist');
+    //  close modal
+    cy.get('app-project-list app-message-modal chef-button:first').click();
+
+    // delete rule
+    cy.request({
+      auth: { bearer: adminIdToken },
+      method: 'DELETE',
+      url: `/apis/iam/v2/projects/${customWithoutPolsProjectID}/rules/${rule.id}`
```

Comment:
Should we check for the "Project UPdate" banner?

Reply:
(I know we don't want to test the functionality of the banner here, however, and we could remove when we add the fuller update tests)

Reply:
sure